### PR TITLE
Group tests in a more standard structure, add code style linting

### PR DESCRIPTION
**Pull request by kattrali**
* Reformat tests as `unittest` cases
* Split up sanitization tests to more easily check for breakage
* Simplify string sanitization
* Lint test code

### CODE REVIEW

File: tests/test_configuration.py [hunk 1]
```diff
@@ -1,67 +1,69 @@
...
-    os.environ["DYNO"] = "YES"
-    c = Configuration()
-    assert(c.hostname == None)
+class TestConfiguration(unittest.TestCase):
+
+    def test_get_endpoint(self):
+        # Test default endpoint with ssl
+        c = Configuration()
+        c.use_ssl = True
+        eq_(c.get_endpoint(), "https://notify.bugsnag.com")
```
Comment (kylef) [inline tests/test_configuration.py]: Why not, `self.assertEqual`?

File: tests/test_configuration.py [hunk 2]
```diff
@@ -1,67 +1,69 @@
...
+        c.use_ssl = False
+        eq_(c.get_endpoint(), "http://notify.bugsnag.com")
+
+        # Test custom endpoint
+        c = Configuration()
+        c.use_ssl = False
+        c.endpoint = "localhost:1234"
+        eq_(c.get_endpoint(), "http://localhost:1234")
+
+    def test_environment_defaults(self):
```
Comment (kylef) [inline tests/test_configuration.py]: Perhaps could be better named:

```python
def test_reads_api_key_from_environ(self):
```

File: tests/test_wsgi.py [hunk 3]
```diff
@@ -14,138 +16,140 @@ class SentinalError(RuntimeError):
...
+                        user={"id": "5", "email":
+                              "me@cirw.in", "name": "conrad"})
+                raise SentinalError("oops")
 
-    eq_(deliver.call_count, 1)
-    payload = deliver.call_args[0][0]
-    eq_(payload['events'][0]['user']['id'], '5')
+        app = TestApp(BugsnagMiddleware(CrashAfterSettingUserId))
 
+        assert_raises(SentinalError, lambda: app.get('/beans'))
```
Comment (kylef) [inline tests/test_wsgi.py]: I'd use the unittest way for this:

```python
with self.assertRaises(SentinalError):
    app.get('/beans')
```

File: tests/test_configuration.py (hunk 1 above)
Comment (kattrali) [inline tests/test_configuration.py]: There were a few uses of `eq_` already and it helped with line length, though I agree that `self.assertEqual` is more readable.

File: tests/test_configuration.py (hunk 2 above)
Comment (kattrali) [inline tests/test_configuration.py]: Agreed.

File: tests/test_wsgi.py (hunk 3 above)
Comment (kattrali) [inline tests/test_wsgi.py]: Makes sense. Also just noticed the typo.

**Pull request merged**